### PR TITLE
Feat: Add cart feature

### PR DESCRIPTION
**Pull request by Vansh353**

### CODE REVIEW

File: controllers/cart_controller.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from dtos.cart_models import AddToCart, GetCartResponse,CartItem
+from helpers.api_helper import APIHelper
+
+
+
+# Add to Cart API
+def add_to_cart(request:AddToCart,token):
+    with engine.connect() as db:
+      
+        existing_cart = db.execute(select(cart_table).where(cart_table.c.product_id == request.product_id).where(cart_table.c.user_id == token.id)).fetchone()
```
Comment (niyati-doshi20) [inline controllers/cart_controller.py]: Why have you used 2 times where condition?

File: controllers/cart_controller.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            return APIHelper.send_success_response(successMessageKey="Cart Updated Successfully")
+        new_cart = db.execute(cart_table.insert().values(
+            user_id=token.id, product_id=request.product_id, product_quantity=request.quantity))
+        db.commit()        
+        return APIHelper.send_success_response(successMessageKey="Product Added to Cart Successfully")
+
+
+def remove_from_cart(product_id,token):
+    with engine.connect() as db:
+        existing_cart = db.execute(select(cart_table).where(cart_table.c.product_id == product_id).where(cart_table.c.user_id == token.id)).fetchone()
```
Comment (niyati-doshi20) [inline controllers/cart_controller.py]: Dont use 2 times where condition.

File: controllers/cart_controller.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      
+        existing_cart = db.execute(select(cart_table).where(cart_table.c.product_id == request.product_id).where(cart_table.c.user_id == token.id)).fetchone()
+        if existing_cart:
+            db.execute(cart_table.update().where(cart_table.c.id == existing_cart.id).values(quantity=existing_cart.quantity+request.quantity))
+            db.commit()
+            return APIHelper.send_success_response(successMessageKey="Cart Updated Successfully")
+        new_cart = db.execute(cart_table.insert().values(
+            user_id=token.id, product_id=request.product_id, product_quantity=request.quantity))
+        db.commit()        
+        return APIHelper.send_success_response(successMessageKey="Product Added to Cart Successfully")
```
Comment (niyati-doshi20) [inline controllers/cart_controller.py]: Provide Response data of cart after add or updated in cart

File: controllers/cart_controller.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                    rating = round(db.execute(select(func.coalesce(func.avg(review_table.c.rating), 0)).where(review_table.c.product_id == row.product_id)).scalar(), 1)
+                    cart_item = CartItem(
+                        product_id=row.product_id,
+                        product_name=product.name,
+                        product_category=product.category,
+                        price=product.price,
+                        rating=rating,
+                        discount=product.discount,
+                        quantity=row.product_quantity,  # Add quantity from cart_table
+                        amount=product.price * row.product_quantity  # Add amount from cart_table
```
Comment (niyati-doshi20) [inline controllers/cart_controller.py]: How discount will be applied here in amount?

Comment (Vansh353) [reply controllers/cart_controller.py]: i am doing that feat by decrement the price

File: controllers/cart_controller.py (hunk 3 above)
Comment (Vansh353) [reply controllers/cart_controller.py]: okay

File: controllers/cart_controller.py (hunk 2 above)
Comment (Vansh353) [reply controllers/cart_controller.py]: removed that

File: controllers/cart_controller.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+
+def remove_from_cart(product_id,token):
+    with engine.connect() as db:
+        existing_cart = db.execute(select(cart_table).where(cart_table.c.product_id == product_id).where(cart_table.c.user_id == token.id)).fetchone()
+        if existing_cart:
+            db.execute(cart_table.delete().where(cart_table.c.id == existing_cart.id))
+            db.commit()
+            return APIHelper.send_success_response(successMessageKey="Product Removed from Cart Successfully")
+        return APIHelper.send_error_response(errorMessageKey="Product Not in Cart")
```
Comment (niyati-doshi20) [inline controllers/cart_controller.py:66]: Proper response message

**Pull request merged**